### PR TITLE
Reduce duplication in Esirkepov deposition

### PR DESCRIPTION
**Pull request by RemiLehe**
This fuses, as much as possible, the 2D and 3D versions of the Esirkepov deposition.

### CODE REVIEW

File: Source/Particles/Deposition/CurrentDeposition.H [hunk 1]
```diff
@@ -251,27 +251,34 @@ void doEsirkepovDepositionShapeN (const Real * const xp, const Real * const yp,
...
+
+#if (AMREX_SPACEDIM == 3)
     const Real invdtdx = 1.0/(dt*dx[1]*dx[2]);
     const Real invdtdy = 1.0/(dt*dx[0]*dx[2]);
     const Real invdtdz = 1.0/(dt*dx[0]*dx[1]);
+#elif (AMREX_SPACEDIM == 2)
+    const Real invdtdx = 1.0/(dt*dx[2]);
+    const Real invdtdz = 1.0/(dt*dx[0]);
+    const Real invvol = 1.0/(dx[0]*dx[2]);
+#endif
```
Comment (MaxThevenet) [inline Source/Particles/Deposition/CurrentDeposition.H]: Minor comment: `invvol` `invdtdx` `invdtdy` `invdtdz` are each used only once in this function. Should it be recomputed for each particle instead?

Comment (MaxThevenet) [reply Source/Particles/Deposition/CurrentDeposition.H]: Disregard this comment.

**Comment by MaxThevenet** [comment]
May I merge this PR?

**Comment by RemiLehe** [comment]
We could maybe let @dpgrote confirm that he agrees with it.

**Pull request merged**